de.dlr.sc.virsat.model/src/de/dlr/sc/virsat/model/dvlm/tree/ITreeTraverserMatcher.java

```diff
@@ -0,0 +1,57 @@
+/*******************************************************************************
+ * Copyright (c) 2008-2019 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package de.dlr.sc.virsat.model.dvlm.tree;
+
+import java.util.Collection;
+
+
+/**
+ * Interface for plugging different logic into {@link BeanStructuralTreeTraverser}
+ * For example for building a tree consisting only of nodes with a certain Category attached
+ * @param <TYPE> The type for which the matcher should apply
+ */
+public interface ITreeTraverserMatcher<TYPE> {
+
+	/**
+	 * Checks if the passed treeNode bean matches the desired criteria
+	 * @param treeNode node bean to check
+	 * @return true if this node matches, false otherwise
+	 */
+	boolean isMatching(TYPE treeNode);
+	
+	/**
+	 * Processes the found match
+	 * @param treeNode a matching tree node
+	 * @param matchingParent the closest parent of the node that also matches the criteria.
+	 *        Can be null if the node does not have a matching parent
+	 */
+	void processMatch(TYPE treeNode, TYPE matchingParent);

```

Comment:
Shall we also add tree level and match level as parameters here?

Reply:
I thought about that one as well, but wanted to keep it to as minimal change as needed. thus i would argues for no, for the moment.